sdk/src/beta9/cli/deployment.py

```diff
@@ -112,13 +113,13 @@ def create_deployment(service: ServiceClient, name: str, entrypoint: str):
 
     module = importlib.import_module(module_name)
 
-    user_func = getattr(module, func_name, None)
+    user_func: Optional[DeployableMixin] = getattr(module, func_name, None)
     if user_func is None:
         terminal.error(
             f"Invalid handler function specified. Make sure '{module_path}' contains the function: '{func_name}'"
         )
 
-    if not user_func.deploy(name=name):  # type:ignore
+    if not user_func.deploy(name=name, context=service._config):  # type: ignore
```

Comment:
why wasn't this required before?

Reply:
It never worked before for the `deploy` command, just for the other commands e.g. `task list` etc. This is because the deploy command is instantiated via the BaseAbstraction class whereas everything else in the CLI uses the ServiceClient gRPC stub wrapper. 

The ServiceClient is created at the CLI entrypoints where we use the `@pass_service_client` decorator which does look for `--context` and if it finds it, uses it to configure channel.

Basically, BaseAbstraction doesn't deal with contexts unless we provide it like we do in the ServiceClient class.

Reply:
Makes sense 👍